core/runtime/src/main/java/io/quarkus/runtime/configuration/ProfileManager.java

```diff
@@ -52,6 +53,10 @@ public static String getActiveProfile() {
             if (profile != null) {
                 return profile;
             }
+            profile = System.getenv(QUARKUS_TEST_PROFILE_ENV);
```

Comment:
Did you try with https://github.com/quarkusio/quarkus/pull/28091/? I believe it should work, because we added the profile configuration key to each `LaunchMode`, and then we remap it to the expected profile key in `ConfigUtils`, so it should look in any of the main sources.